src/engine/sim_engine/pool.rs

```diff
@@ -758,6 +758,26 @@ impl Pool for SimPool {
     fn metadata_version(&self) -> StratSigblockVersion {
         StratSigblockVersion::V2
     }
+
+    fn current_fs_metadata(&self, fs_name: Option<&str>) -> StratisResult<String> {
+        self.filesystems
+            .iter()
+            .filter_map(|(name, uuid, fs)| {
+                if fs_name.map(|n| *n == **name).unwrap_or(true) {
+                    Some(serde_json::to_string(&fs.record(name, *uuid)).map_err(|e| e.into()))
+                } else {
+                    None
+                }
+            })
+            .collect::<StratisResult<Vec<_>>>()
+            .map(|v| v.join("\n"))
+    }
```

Comment:
If I'm understanding this correctly, this means if fs_name is `None` it will return a bunch of JSON blobs separated by newlines for all filesystems present in the pool. Wouldn't it be better to put this all into a JSON object mapping UUIDs to metadata? I'm unaware of any JSON parser that can handle multiple separate blobs of JSON at once.

Reply:
@jbaublitz That's exactly what it does. I was of two minds about this, but then I found that ```jq``` is fine with distinct blobs of JSON and I grew more satisfied with it.
```
$ cat junk.txt | jq
{}
{
  "key": 2
}
```

But your proposed solution is likely better.

Reply:
I don't know of any JSON parser in a programming language like Python that will do the same unfortunately.